config/config_root.rb

```diff
@@ -0,0 +1,63 @@
+require 'pathname'
+require 'dotenv'
+
+module ConfigRoot
+  # make these methods also callable directly
+  # see for discussion on extend self vs module_function:
+  # https://github.com/bbatsov/ruby-style-guide/issues/556
+  extend self
+
+  # The app's configuration root directory
+  # @return [Pathname] path to configuration root
+  def config_root
+    Pathname.new(ENV["OOD_APP_CONFIG"] || "/etc/ood/config/apps/dashboard")
+  end
+
+
+  # Load the dotenv local files first, then the /etc dotenv files and
+  # the .env and .env.production or .env.development files.
+  #
+  # Doing this in two separate loads means OOD_APP_CONFIG can be specified in
+  # the .env.local file, which will specify where to look for the /etc dotenv
+  # files. The default for OOD_APP_CONFIG is /etc/ood/config/apps/myjobs and
+  # both .env and .env.production will be searched for there.
+  def load_dotenv_files
+    Dir.chdir app_root do
+      # .env.local first, so it can override OOD_APP_CONFIG
+      Dotenv.load(*dotenv_local_files) unless dotenv_local_files.empty?
```

Comment:
Due to the strange complexity this introduces just to get at `OOD_APP_CONFIG`, why don't we just introduce a new env file such as...

```ruby
Dotenv.load([app_root.join(".env.config")])
```

or

```ruby
Dotenv.load([app_root.join(".env.all.local")])
```

or 

```ruby
Dotenv.load([app_root.join(".env.super.local")])
```

This way it isn't a *possible* breaking change (for those super rare edge cases) like:

```bash
# .env
A=${B}
```

```bash
# .env.local
B="should be local to this file only"
```

It should return:

```ruby
puts ENV["A"] #=> ""
```

but instead it returns:

```ruby
puts ENV["A"] #=> "should be local to this file only"
```

Reply:
This will really only ever be used by Dashboard developers, so I don't think it will be a problem.